src/astronomy/moshier/util.js

```diff
@@ -204,7 +204,7 @@ $ns.util.showcor = function (p, dp, result) {
  * from input equatorial rectangular unit vector.
  * Output vector pol[] contains R.A., Dec., and radius.
  */
-$ns.util.showrd = function (p, pol, result) {
```

Comment:
Normally functions like these with a variable passed at the end which looks like its never used is basically a Recursive function that calls self and uses the previous value if at all it was passed. I assume you are 100% sure that this method is not self calling, if thats the case, let me know otherwise, its wise to double check for any recursion.

also, I have seen changes like this at multiple places, it will be nice to see similar recursion there as well.

Reply:
I definitely checked for any uses of such parameters before removing them. But I also welcome a second pair of eyes. To search for uses of this function, I searched all the source files for the string "showrd" - the name of the function. The search results are not overwhelming, so I think all the uses of this function can quickly be reviewed. I refactored a few other functions as well, similarly removing parameters where possible. I tried my best to provide such changes in incremental fashion, so the stability of the code can be evaluated from one commit to the next. Each commit in this pull request should leave the code in a stable condition with one exception: the last two commits ("type structure refactor" and "type structure refactor part 2" belong together)